agent/eventhandler/task_handler.go

```diff
@@ -47,13 +47,18 @@ type TaskHandler struct {
 	tasksToEvents map[string]*eventList
 	// tasksToEventsLock for locking the map
 	tasksToEventsLock sync.RWMutex
+	// batchMap is used to collect container events
+	// between task transitions
+	batchMap     map[string][]api.ContainerStateChange
```

Comment:
Can you rename this as `tasksToContainerStates` ? If you do that, you can rename the lock as well